docs/source/getting_started/local_development.md

```diff
@@ -0,0 +1,171 @@
+# Local development
+
+You can run the examples and tests locally on an M1 mac.
+
+## Installation
+
+We use the [uv](https://docs.astral.sh/uv/) package manager to manage the
+installation of the plugin and its dependencies. `uv` provides advanced
+dependency resolution which is required to properly install dependencies like
+`vllm` without overwriting critical dependencies like `torch`.
+
+1. Clone vllm-spyre
+
+   ```sh
+   git clone https://github.com/vllm-project/vllm-spyre.git
+   cd vllm-spyre
+   ```
+
+1. Install uv
+  
+   ```sh
+   pip install uv
+   ```
+  
+1. Create a new env
+
+   ```sh
+   uv venv --python 3.12 --seed
+   ```
+
+1. Activate it
+  
+   ```sh
+   source .venv/bin/activate
+   ```
+
+1. Install `vllm-spyre` locally with dev (and optionally lint) dependencies
+  
+   ```sh
+   uv sync --frozen
+   ```
+  
+   or also with lint:
+  
+   ```sh
+   uv sync --frozen --group lint
+   ```
+  
+   **Note: `--group dev` is enabled by default**
```

Comment:
You could use nice-looking ["admonitions"](https://myst-parser.readthedocs.io/en/latest/syntax/admonitions.html#admonition-types) for notes with more context like this one.